src/foam/lib/query/StringParser.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.lib.query;
+
+import foam.lib.json.ASCIIEscapeParser;
+import foam.lib.json.UnicodeParser;
+import foam.lib.parse.PStream;
+import foam.lib.parse.Parser;
+import foam.lib.parse.ParserContext;
+
+public class StringParser implements Parser {
+  public final static char ESCAPE = '\\';
+
+  public StringParser() {
+  }
+
+  public PStream parse(PStream ps, ParserContext x) {
+
+    if ( !ps.valid() )  return null;
+    char delim = ps.beforeHead();
+
+    delim = ( delim == '=' || delim == ':' || delim == '(' ) ? ' ' : delim;
+
+    if ( Character.isLetter(delim) ) {
+      delim = ' ';
+    }
+
+    char lastc = delim;
+
+    // TODO: use thread-local SB instead to avoid generating garbage
+    StringBuilder sb = new StringBuilder();
+
+    while ( ps.valid() ) {
+      char c = ps.head();
+      if ( c == ')' ) {
+        ps.decrement();

```

Comment:
Instead of decrementing, you can just hold on to the previous pstream in a different variable.

Look at how foam.lib.json.StringParser stores the pstream in a separate variable named tail, and updates ps = tail at the end of its inner loop.